packages/reactive-element/src/decorators/query.ts

```diff
@@ -63,9 +63,11 @@ export function query(selector: string, cache?: boolean) {
               key as string
             ] = this.renderRoot?.querySelector(selector);
           }
-          return (this as unknown as {[key: string]: Element | null})[
-            key as string
-          ];
+          return (
+            (this as unknown as {[key: string]: Element | null})[
+              key as string
+            ] ?? null
```

Comment:
should we initialize the cache property to `null` so that we avoid later hidden class changes and then check for `null` instead of `undefined` above?

Reply:
There's no real place to do that right? You can't do it at decorator time since it's an own property on the instance. And there's not a good initializer spot to do it before the accessor is run the first time ... 🤷 .

Reply:
This wasn't quite correct. The intended behavior is to always cache the first result and that the result must be either null or the found element.